### PR TITLE
roachprod: upgrade to a newer ubuntu image

### PR DESCRIPTION
**Pull request by rickystewart**
Also stop and mask some services on the `roachprod` machines to reduce
pressure and minimize the impact of a pervasive clock sync bug.

Release note: None

### CODE REVIEW

**Comment by rickystewart** [comment]
Still waiting for CI, but local runs look good, so it's probably ready for review.

**Review by rail — APPROVED** [review]

**Comment by rafiss** [comment]
did you kick off a nightly run for this branch? i noticed a bunch of new failures come in that point at this PR

https://github.com/cockroachdb/cockroach/issues/63522
https://github.com/cockroachdb/cockroach/issues/63517
https://github.com/cockroachdb/cockroach/issues/63516
https://github.com/cockroachdb/cockroach/issues/63505
https://github.com/cockroachdb/cockroach/issues/63504

haven't had a chance to look yet to see what the root cause is / if it's just noise

**Comment by rickystewart** [comment]
Yes, I manually kicked off a nightly run: https://teamcity.cockroachdb.com/viewLog.html?buildId=2873725&

**Comment by rickystewart** [comment]
the build config checks out `master` before it does anything so the build I ran yesterday didn't actually test any of my changes at all 🙃

**Comment by rickystewart** [comment]
take 2: https://teamcity.cockroachdb.com/viewQueued.html?itemId=2878303

**Comment by rickystewart** [comment]
A couple observations: 

1. The `clock` tests are now failing since the Ubuntu upgrade breaks [`bumptime`](https://github.com/cockroachdb/jepsen/blob/master/cockroachdb/resources/bumptime.c). This is evidently due to a [kernel upgrade](https://stackoverflow.com/questions/35353766/how-to-use-settimeofday2) that limits what you can call `settimeofday` with.
2. `disk-stalled` tests are failing due to `python` missing. (Unclear if we can just replace `python` with `python3`.)

**Comment by rickystewart** [comment]
https://github.com/cockroachdb/jepsen/pull/25 will take care of the `bumptime` thing, so this can be merged when that hits `master`.

**Comment by tbg** [comment]
bors r=rail

failed on an unrelated unit test

**Comment by tbg** [comment]
Essential CI failed, but without a failed subtest. I saw this before, it can happen when a worker machine gets preempted. One more go!

bors r=rail

**Pull request merged**
